homeassistant/components/input_number.py

```diff
@@ -122,7 +121,7 @@ def _cv_input_number(cfg):
     return True
 
 
-class InputNumber(Entity):
+class InputNumber(RestoreEntity):
```

Comment:
this doesn't make it look like a mixin?

Reply:
Well, there's no real mixins in Python, just actual multiple inheritance, but I can change it to `InputNumber(Entity, RestoreEntity)` if we'd prefer that.

Reply:
Yes. RestoreEnttiy should not inherit from Entity, it should only include the parts that are part of the mixin.

Reply:
RestoreEntity still needs to inherit from Entity since it needs to access the `hass` and `entity_id` attributes from `Entity`.

Reply:
As a mixin we can assume we are mixed into an Entity class. We shouldn't inherit ourselves from it because that will cause confusing things if the order of inheritance. It could now be that the Entity.state will override the Light.state property again because this mixin pulls in Entity.

Reply:
OK, I tried stripping out the base class, however it isn't as simple as I thought. Because we're overriding `async_added_to_hass`, we can't call the super method if we don't declare the inheritance properly.

I also think there's a misunderstanding here about how multiple inheritance works in Python. No matter how many classes an entity subclasses that themselves subclass Entity, Entity is still only pulled into the inheritance tree once. Multiple classes that subclass Entity won't clobber each other in the way you're describing. The Python MRO is already aware of this potential complication and handles it for us.